Initial support for I2C on STM32 H7

gmarull opened the pull request:
- Enable I2C driver for H7 series
- Enable I2C on NUCLEO-H743ZI board.

gmarull committed:
drivers: i2c: stm32: add support for H7 series

Enable I2C driver for STM32 H7 series.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
boards: arm: nucleo_h743zi: enable I2C support

Enable I2C support, including for the Arduino R3 connector.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

MaureenHelm commented:
Please rebase and fix the merge conflicts

FRASTM commented on drivers/pinmux/stm32/pinmux_stm32h7.h:
```diff
@@ -50,19 +54,41 @@
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
 #define STM32H7_PINMUX_FUNC_PB6_LPUART1_TX    \
-	(STM32_PINMUX_ALT_FUNC_6 | STM32_OPENDRAIN_PULLUP)
+	(STM32_PINMUX_ALT_FUNC_8 | STM32_OPENDRAIN_PULLUP)
```
This is an issue, not related to I2C, which could be fixed in another commit

gmarull replied:
@FRASTM done in 3077a04

gmarull committed:
drivers: pinmux: stm32: fix PB6 LPUART1 alt function on H7

LPUART1 TX alternate function number for PB6 was not correct in the
pinmux settings for H7 series.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull commented:
@MaureenHelm done
